use/mobile-gaming-example.md

```diff
@@ -0,0 +1,540 @@
+
+For pipelines that read unbounded game data from a streaming source, the data source sets the intrinsic [timestamp](/learn/programming-guide/#pctimestamps) for each PCollection element to the appropriate event time.
+
+The Mobile Game example pipelines vary in complexity, from simple batch analysis to more complex pipelines that can perform real-time analysis and abuse detection. This section walks you through each example and demonstrates how to use Beam features like windowing and triggers to expand your pipeline's capabilites.
+
+## UserScore: Basic Score Processing in Batch
+
+The `UserScore` pipeline is the simplest example for processing mobile game data. `UserScore` determines the total score per user over a finite data set (for example, one day's worth of scores stored on the game server). Pipelines like `UserScore` are best run periodically after all relevant data has been gathered. For example, `UserScore` could run as a nightly job over data gathered during that day.
+
+> **Note:** See [UserScore on GitHub](https://github.com/apache/incubator-beam/blob/ee6ad2fe416cd499aa1b6e2a51aa64da0805cc5c/examples/java8/src/main/java/org/apache/beam/examples/complete/game/UserScore.java) for the complete example pipeline program.
+
+### What Does UserScore Do?
+
+In a day's worth of scoring data, each user ID may have multiple records (if the user plays more than one instance of the game during the analysis window), each with their own score value and timestamp. If we want to determine the total score over all the instances a user plays during the day, our pipeline will need to group all the records together per individual user.
+
+As the pipeline processes each event, the event score gets added to the sum total for that particular user.
+
+`UserScore` parses out only the data that it needs from each record, specifically the user ID and the score value. The pipeline doesn't consider the event time for any record; it simply processes all data present in the input files that you specify when you run the pipeline.
+
+> **Note:** To use the `UserScore` pipeline effectively, you'd need to ensure that you supply input data that has already been grouped by the desired event time period — that is, that you specify an input file that only contains data from the day you care about.
+
+`UserScore`'s basic pipeline flow does the following:
+
+1. Read the day's score data from a file stored in Google Cloud Storage.
+2. Sum the score values for each unique user by grouping each game event by user ID and combining the score values to get the total score for that particular user.
+3. Write the result data to a [Google Cloud BigQuery](https://cloud.google.com/bigquery/) table.
+
+The following diagram shows score data for a several users over the pipeline analysis period. In the diagram, each data point is an event that results in one user/score pair:
```

Comment:
I suggest we explain more clearly what the two axes mean and how a specific point might have a given position. I find the 'ideal' line (x=y) very helpful here.